Watch rows/cols and resize the term and send sigwinch signal to exec

jwforres opened the pull request.

jwforres committed:
Watch rows/cols and resize the term and send sigwinch signal to exec

jwforres commented:
@ncdc try this with your change to the API, if you update the rows/cols in the example index file it should resize the term and send the signal

@stefwalter decided to make it up to whoever is using the container terminal to decide when / if to resize it by updating the rows cols we pass into the scope

jwforres commented:
updated, fixed a bug

stefwalter commented:
Nice work.

Is there a link to the kubernetes pull request to test this against? I'd like to test this before merging.

ncdc commented:
I'm still working on it - the WIP is here: https://github.com/ncdc/kubernetes/pull/7

ncdc commented:
And yes, I think waiting for the kube PR to merge makes sense

ncdc commented on dist/container-terminal.js:
```diff
@@ -213,6 +215,15 @@
+                          var rows = newValues[1] || defaultRows;
+                          term.resize(cols, rows);
+                          if (ws && ws.readyState === 1) {
+                            ws.send("4" + window.btoa('{"Width": cols, "Height": rows}'));                          
```
Need to use the actual values for cols and rows 😄 

``` javascript
ws.send("4" + window.btoa('{"Width":' + cols + ',"Height":' + rows + '}'));
```

jwforres replied:
Hah whoops, good catch. Will update that in the morning

ncdc replied:
Cool. With this change, it works!

ncdc commented:
Kube PR: https://github.com/kubernetes/kubernetes/pull/25273

jwforres commented:
@ncdc saw the API PR is trying to get merged, did anything change or is the impl in this PR still correct for sending the signal

ncdc commented:
@jwforres it should still be the same but I'll doublecheck to make sure this still works

ncdc commented:
@jwforres yes it works, but you still need to do https://github.com/kubernetes-ui/container-terminal/pull/7#discussion_r61191752

jwforres commented:
rebased and fixed https://github.com/kubernetes-ui/container-terminal/pull/7#discussion_r61191752

should be ready to go as soon as the API change gets merged

jwforres commented:
cc @spadgett
